vertical-pod-autoscaler/recommender/clients/cluster_feeder.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clients
+
+import (
+	"github.com/golang/glog"
+	vpa_api "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/client/clientset/versioned/typed/poc.autoscaling.k8s.io/v1alpha1"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/recommender/model"
+)
+
+// ClusterStateFeeder can update state of ClusterState object.
+type ClusterStateFeeder interface {
+	// Subscribe saves a pointer to a ClusterState object which will be updated.
+	Subscribe(cluster *model.ClusterState)
```

Comment:
Why do you want to have a subscriber model here?

Reply:
This is just to semantically divide data providers from data consumers.

Data providers are all passed in "constructor", and data consumer must "subscribe" to be updated. For me, it makes the dataflow clearer, especially if there are more data producers in the future.